File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/SetTickIntervalOfTaskCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class SetTickIntervalOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    constructor(nodeId, tickInterval)
    {
        super();
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("SetTickIntervalOfTaskCommandMsg");
        this.jsonMessage["nodeId"] = nodeId;
        this.jsonMessage["tickInterval"] = tickInterval;
    }
}

module.exports = {SetTickIntervalOfTaskCommandMsg};

File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/SetTickIntervalOfTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class SetTickIntervalOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    Long nodeId;
    Integer tickInterval;

    public SetTickIntervalOfTaskCommandMsg(Long nodeId, Integer tickInterval)
    {
        this.nodeId = nodeId;
        this.tickInterval = tickInterval;
    }

    public Long getNodeId()
    {
        return nodeId;
    }

    public Integer getTickInterval()
    {
        return tickInterval;
    }
}
